docarray/array/storage/qdrant/getsetdel.py

```diff
@@ -0,0 +1,118 @@
+from typing import List, Iterable, Iterator
+
+import scipy.sparse
+from qdrant_client import QdrantClient
+from qdrant_openapi_client.exceptions import UnexpectedResponse
+from qdrant_openapi_client.models.models import PointIdsList, PointsList, ScrollRequest
+
+from docarray import Document
+from docarray.array.storage.base.getsetdel import BaseGetSetDelMixin
+from docarray.types import ArrayType
+
+SCROLL_BATCH_SIZE = 64
+
+
+class GetSetDelMixin(BaseGetSetDelMixin):
+
+    @property
+    def client(self) -> QdrantClient:
+        raise NotImplementedError()
+
+    @property
+    def serialization_config(self) -> dict:
+        raise NotImplementedError()
+
+    @property
+    def n_dim(self) -> int:
+        raise NotImplementedError()
+
+    @property
+    def collection_name(self) -> str:
+        raise NotImplementedError()
+
+    def _embedding_to_array(self, embedding: ArrayType) -> List[float]:
+        if embedding is None:
+            embedding = [0] * self.n_dim
+        elif isinstance(embedding, scipy.sparse.spmatrix):
+            embedding = embedding.toarray().tolist()
+        else:
+            from ....math.ndarray import to_numpy_array
+            embedding = to_numpy_array(embedding).tolist()
+
+        return embedding
+
+    def _upload_batch(self, docs: Iterable[Document]):
+        self.client.http.points_api.upsert_points(
+            name=self.collection_name,
+            wait=True,
+            point_insert_operations=PointsList(
+                points=[self._document_to_qdrant(doc) for doc in docs]
+            )
+        )
+
+    def _qdrant_to_document(self, qdrant_record: dict) -> 'Document':
+        return Document.from_base64(
+            qdrant_record['payload']['_serialized'], **self.serialization_config
+        )
+
+    def _document_to_qdrant(self, doc: Document, _id: str = None) -> dict:
+        return dict(
+            id=_id or doc.id,
```

Comment:
Alright, the current behaviour of DocumentArray is to rely on `value.id` as the new ID and `_id` as the existing, old ID.
So basicly, in case both IDs are different, this should change the ID of the Document in the storage backend.